cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaListenersReconciler.java

```diff
@@ -116,18 +116,19 @@ public KafkaListenersReconciler(
      * The main reconciliation method which triggers the whole reconciliation pipeline. This is the method which is
      * expected to be called from the outside to trigger the reconciliation.
      *
+     * @param replicas replicas
      * @return  Future which completes when the reconciliation completes. It contains the result of the reconciliation
      *          which contains the collected addresses, prepared listener statuses etc.
      */
-    public Future<ReconciliationResult> reconcile()    {
-        return services()
-                .compose(i -> routes())
-                .compose(i -> ingresses())
-                .compose(i -> internalServicesReady())
-                .compose(i -> loadBalancerServicesReady())
-                .compose(i -> nodePortServicesReady())
-                .compose(i -> routesReady())
-                .compose(i -> ingressesReady())
+    public Future<ReconciliationResult> reconcile(int replicas)    {
```

Comment:
Assuming the replicas are actually needed in all these method, why not pass it through constructor?